### PR TITLE
Initial migration to registerMethod2

### PR DESCRIPTION
**Pull request by pentacular**
Starts work on separating out the flexible and simple api interfaces.

### CODE REVIEW

File: api/core/evaluate.js [hunk 1]
```diff
@@ -25,11 +25,7 @@ export const evaluate = async (ecmascript, { api, path }) => {
     );
     // Add import to make import.meta.url available.
     const op = await builder({ ...api, import: { meta: { url: path } } });
-    // Retry until none of the operations block.
-    // Note: This retry mechanism should be obsolete now.
```
Comment (BarbourSmith) [inline api/core/evaluate.js:29]: 👍

File: api/shape/Plan.js [hunk 2]
```diff
@@ -46,12 +46,12 @@ export const computeSides = (c1, c2, sides, zag = 0.01) => {
 export const zagSides = Shape.registerMethod2(
   'zagSides',
   ['number', 'number'],
-  (diameter = 1, zag = 0.01) => 1 / toSidesFromZag(diameter, zag)
+  (diameter = 1, zag = 0.01) => toSidesFromZag(diameter, zag)
```
Comment (BarbourSmith) [inline api/shape/Plan.js:49]: What does this change mean? Why was it 1/ before...ah was the number of segments?

File: api/shape/destructure.js [hunk 3]
```diff
@@ -333,7 +379,35 @@ export const destructure2 = async (shape, input, ...specs) => {
...
+          } else if (Shape.isArray(value) && value.every(Shape.isSegment)) {
+            out.push(...value);
+          } else {
+            rest.push(arg);
+          }
+        }
+        output.push(out);
+        break;
+      }
+      case 'coordinateLists': {
```
Comment (BarbourSmith) [inline api/shape/destructure.js:406]: Has "coordinates" fully replaced "points"?

File: api/shape/nth.js [hunk 4]
```diff
@@ -1,35 +1,33 @@
...
-        if (nth < 0) {
-          nth = candidates.length - nth;
-        }
-        const candidate = candidates[nth];
-        if (candidate === undefined) {
-          group.push(Empty());
-        } else {
-          group.push(candidate);
-        }
+  ['input', 'numbers'],
```
Comment (BarbourSmith) [inline api/shape/nth.js:8]: How do these work? Is this for documentation?

**Review by BarbourSmith — APPROVED** [review]
Wow, this is a big pull request :)

Looks great to me!

File: api/shape/Plan.js (hunk 2 above)
Comment (pentacular) [reply api/shape/Plan.js:49]: I managed to confuse it with zagStep :)

File: api/shape/nth.js (hunk 4 above)
Comment (pentacular) [reply api/shape/nth.js:8]: This tells it how to do the magical decoding stuff.

This means, "look for an input shape first, then look for numbers".

But, yeah, it should also be used to generate documentation and UI automatically.

These then get passed in with the corresponding arguments to the actual function (which should move out of here and into geometry)

File: api/shape/destructure.js (hunk 3 above)
Comment (pentacular) [reply api/shape/destructure.js:406]: I've started using coordinates for things like [1, 2, 3] as opposed to Point()

**Pull request merged**